File: Daily Test/C++/Invert Sign & Evaluate Polynomial Expression 11.11.2025.cpp
#include <iostream>
#include <string>
#include <cctype>
using namespace std;

void replace_chars(string &str) {
    for (char &ch : str) {
        if (ch == '+') ch = '_';
        else if (ch == '-') ch = '+';
    }
    for (char &ch : str) {
        if (ch == '_') ch = '-';
    }
}

int main() {
    string b, c;
    getline(cin, b);
    getline(cin, c);
    string d = b;
    replace_chars(d);
    cout << d << endl;

    // Simple evaluator for "number x number" type input
    int res = 0, i = 0, xval = stoi(c);
    while (i < d.size()) {
        if (isdigit(d[i])) {
            int coeff = d[i] - '0';
            i++;
            if (d[i] == 'x') {
                res += coeff * xval;
                i++;
            } else {
                res += coeff;
            }
        } else if (d[i] == '+') {
            i++;
        } else if (d[i] == '-') {
            i++;
            if (isdigit(d[i])) {
                int coeff = d[i] - '0';
                i++;
                if (d[i] == 'x') {
                    res -= coeff * xval;
                    i++;
                } else {
                    res -= coeff;
                }
            }
        } else {
            i++;
        }
    }
    cout << res << endl;
    return 0;
}

File: Daily Test/Java/Invert Sign & Evaluate Polynomial Expression 11.11.2025.java
import java.util.Scanner;

public class Main {
    static String replaceChars(String str) {
        str = str.replace('+', '_').replace('-', '+').replace('_', '-');
        return str;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String b = scanner.nextLine().trim();
        String c = scanner.nextLine().trim();
        String d = replaceChars(b);
        System.out.println(d);

        int res = 0;
        int xval = Integer.parseInt(c);
        int i = 0;
        while (i < d.length()) {
            char ch = d.charAt(i);
            if (Character.isDigit(ch)) {
                int coeff = ch - '0';
                i++;
                if (i < d.length() && d.charAt(i) == 'x') {
                    res += coeff * xval;
                    i++;
                } else {
                    res += coeff;
                }
            } else if (ch == '+') {
                i++;
            } else if (ch == '-') {
                i++;
                if (i < d.length() && Character.isDigit(d.charAt(i))) {
                    int coeff = d.charAt(i) - '0';
                    i++;
                    if (i < d.length() && d.charAt(i) == 'x') {
                        res -= coeff * xval;
                        i++;
                    } else {
                        res -= coeff;
                    }
                }
            } else {
                i++;
            }
        }
        System.out.println(res);
        scanner.close();
    }
}